input.zig

```diff
@@ -606,14 +722,33 @@ const State = struct {
         enabled: struct { confined: bool },
     } = .disabled,
     confine_grab: bool = false,
-    disable_input_device: union(enum) {
+
+    query_input_extension: union(enum) {
```

Comment:
Split out the querying of the X input extension to it's own state (`query_input_extension`) so we can share the result with the `listen_to_raw_events` and `disable_input_device` state machines